### PR TITLE
Feature/donation history phase 2

### PR DESCRIPTION
**Pull request by liruchen32**
# Issue
[[會員制]個人贊助紀錄 phase2-前端](https://app.asana.com/0/1203699264568808/1207133511967082/f)

# Notice
~因為贊助收據資料需要跨分頁傳遞，所以使用 `indexeddb` ，但可能需要定期清理，或是開新 api 來給下載頁取資料~
後來發現可以用現有的 api 取得需要的資料

# Dependency
[npm-packages#534](https://github.com/twreporter/twreporter-npm-packages/pull/534)

### CODE REVIEW

File: src/components/member-page/donation/download-page.js [hunk 1]
```diff
@@ -0,0 +1,377 @@
...
+  const [receiptAddress, setReceiptAddress] = useState('')
+  const [receiptPayMethod, setReceiptPayMethod] = useState('')
+  const [cardLastFour, setCardLastFour] = useState('')
+  const [cardType, setCardType] = useState('')
+  const [isLoading, setIsLoading] = useState(true)
+  const jwt = _.get(state, [reduxStateFields.auth, 'accessToken'])
+  const apiOrigin = _.get(state, [reduxStateFields.origins, 'api'])
+
+  const getReceiptData = async () => {
+    const { data: orderResponseData } = await axios.get(
```
Comment (Aylie-Chou) [inline src/components/member-page/donation/download-page.js:190]: Could we add try/catch for error handling of axios

File: src/components/member-page/donation/download-page.js [hunk 2]
```diff
@@ -0,0 +1,377 @@
...
+      ''}${receipt.address_detail || ''}`
+    setReceiptHeader(receipt.header)
+    setReceiptAddress(address)
+    setReceiptPayMethod(payMethod)
+    setCardLastFour(cardInfo.last_four)
+    setCardType(cardInfo.type)
+  }
+  const getPaymentHistory = async () => {
+    // set limit to -1 to get all records
+    const { data: historyResponseData } = await axios.get(
```
Comment (Aylie-Chou) [inline src/components/member-page/donation/download-page.js:215]: same as [this comment](https://github.com/twreporter/twreporter-react/pull/2837/files#r1592915147)

File: src/components/member-page/donation/download-page.js [hunk 3]
```diff
@@ -0,0 +1,377 @@
...
+    getPaymentHistory()
+    setIsLoading(false)
+    window.print()
+  }, [])
+
+  return (
+    <LoadingMask isFetching={isLoading} showSpinner={isLoading}>
+      <Container>
+        <Header>
+          <svg
```
Comment (Aylie-Chou) [inline src/components/member-page/donation/download-page.js:239]: could we store the svg content to another file
(we could use it as jsx with import)

**Review by Aylie-Chou — CHANGES_REQUESTED** [review]
nice work to give birth to download page !
please have a look at comments ~~

File: src/components/member-page/donation/download-page.js (hunk 2 above)
Comment (liruchen32) [reply src/components/member-page/donation/download-page.js]: fixed

File: src/components/member-page/donation/download-page.js (hunk 3 above)
Comment (liruchen32) [reply src/components/member-page/donation/download-page.js]: fixed

**Review by Aylie-Chou — APPROVED** [review]
LGTM

**Pull request merged**